### ISSUE TITLE
Gear language already in use

### ISSUE THREAD

**Issue by zippers** [amlantis-lang/aml-doc#16]
I've been working on a language called Gear, [publicly on github](https://github.com/zippers/gear), for the past two years (see my [first commit](https://github.com/zippers/gear/commit/6185a89c10fe138dcf58ce50a1714342ae2c4d2d) two years ago).  Your copyright regarding the name Gear says 2012 to 2015, but from my research, you only started using the name within the past few months.  You used the name Coral prior to the switch.

As language designers, we strive to conceive unique names for our programming languages.  Imagine my sadness when I saw Gear being adopted elsewhere.  I'm sure you can imagine my frustration.

Since you chose the name Gear relatively recently, I was wondering if you'd be willing to change the name to avoid confusion.

I'm sure you are a good person and wouldn't have knowingly used the name had you known it was taken.  I hope we can come to an understanding.  Thanks for hearing me out.

**Comment by TheMarketka** [comment]
In all honesty, I’m not afraid of people confusing my actively developed project with your 8 months dead project.

**Comment by zippers** [comment]
Thanks for taking the time to respond.

> In all honesty, I’m not afraid of people confusing my actively developed project with your 8 months dead project.

The project is not dead.  It will be receiving a very large update soon.  In the future, please contact the original author about forfeiting their copyright first.

Its sad when large companies like Apple (swift) or Google (go) use existing names without permission.  I can't do much if you decide to continue using Gear, but I hope you would reconsider respecting my copyright.

If my copyright is not respected, then this impacts the trust and integrity of the open source community.

Copyrights are meaningless if not respected.  I urge you to reconsider.

**Comment by TheMarketka** [comment]
> Thanks for taking the time to respond.

You’re welcome. 

> The project is not dead. It will be receiving a very large update soon. In the future, please contact the original author about forfeiting their copyright first.

Informally, are you sure that anyone would see that coming? As far as I could tell (without contacting you), you could be retired or worse, dead. Personally, when I’m looking for an open source software solution, I check its activity, and 8 months no activity = dead, abandoned, inactive. Admittedly, I could have actually contacted you and asked about it, for not doing so I am sorry. 

> Its sad when large companies like Apple (…)

True. Also the reason I abandoned the name `Coral`, because there was already a language named `CORAL 66`, present on Wikipedia and elsewhere, with its own webpage and stuff, and I did not feel like being Apple or Google. However, I already paid for the `gear-lang.org` domain and do not want my money wasted. 

> If my copyright is not respected, then this impacts the trust and integrity of the open source community.

Big words for a little unknown project, don’t you think? The integrity of the open source community will definitely stay intact, I am sure of that. 

If I’m going to change the name (again!), it will be when I find a better suiting name, including an available `*-lang.org` domain, which did not happen yet. I’m open to more suggestions, obviously. If a better name is found, I can also offer you the domain, if you’re interested, it would be a shame to let it go.

**Comment by zippers** [comment]
>Big words for a little unknown project, don’t you think? The integrity of the open source community will definitely stay intact, I am sure of that.

You are correct.  I am a little unknown project.  However, copyright should be respected regardless of size.  I love open source and the community.  I love the mutual respect developers can show one another.  I'm grateful you are engaging with me on this issue.

> Informally, are you sure that anyone would see that coming? As far as I could tell (without contacting you), you could be retired or worse, dead. Personally, when I’m looking for an open source software solution, I check its activity, and 8 months no activity = dead, abandoned, inactive. Admittedly, I could have actually contacted you and asked about it, for not doing so I am sorry.

Gear has been a labor of love for me for the past 2 years.  There is a personal story behind its name and the impact its had on me.  This is why I've reached out to you.  I would like to keep the name if at all possible.

> However, I already paid for the gear-lang.org domain and do not want my money wasted.

I would be willing to purchase the domain from you so you can recoup your loses.

Thanks for engaging with me.

**Comment by TheMarketka** [comment]
Alright, you sound reasonable (and I would love to hear the background story if possible), how about we make a deal: I search for a better name, and when I’ve found one, I’ll let you know and we can transfer the domain name etc. I’ll try to find one as soon as I can, but I can’t really make any guarantees on how long it would take. Would that be ok?

**Comment by zippers** [comment]
> Alright, you sound reasonable (and I would love to hear the background story if possible), how about we make a deal: I search for a better name, and when I’ve found one, I’ll let you know and we can transfer the domain name etc. I’ll try to find one as soon as I can, but I can’t really make any guarantees on how long it would take. Would that be ok?

That would be wonderful.  Here is the story:

Several years ago, when I was in college, I was close to a professor, [Eric Shade](http://people.missouristate.edu/EricShade/).  He was a brilliant professor and fascinated by programming languages.  He helped foster an interest in them in me.

One day, I walked into his office and he was working on the floor inside his desktop computer.  I didn't notice it, but he had a mechanical watch opened on the table next to him.  I accidentally knocked the table and the watch parts went flying into the open machine.

He didn't get mad at me.  He just smiled and I helped him meticulously pull each part out.  As I went to set the parts back on the table, he told me I could throw them away.  Apparently the watch had stopped working long ago.  As I went to throw them away, one of the tiny gears caught my eye.  I ended up pocketing it and throwing away the rest.

After I graduated, I stayed in touch with him.  Unfortunately, he [passed away](http://www.legacy.com/obituaries/news-leader/obituary.aspx?pid=174514292) earlier this year.  All I have left is this tiny gear.
